kotlin/internal/jvm/compile.bzl

```diff
@@ -321,6 +321,45 @@ def _run_kapt_builder_actions(
         kapt_generated_class_jar = kapt_generated_class_jar,
     )
 
+def _run_ksp_builder_actions(
+        ctx,
+        rule_kind,
+        toolchains,
+        srcs,
+        associates,
+        compile_deps,
+        deps_artifacts,
+        annotation_processors,
+        transitive_runtime_jars,
+        plugins):
+    """Runs KSP using the KotlinBuilder tool
+
+    Returns:
+        A struct containing KSP outputs
+    """
+    ksp_generated_java_srcjar = ctx.actions.declare_file(ctx.label.name + "-ksp-kt-gensrc.jar")
+
+    _run_kt_builder_action(
+        ctx = ctx,
+        rule_kind = rule_kind,
+        toolchains = toolchains,
+        srcs = srcs,
+        generated_src_jars = [],
+        associates = associates,
+        compile_deps = compile_deps,
+        deps_artifacts = deps_artifacts,
+        annotation_processors = annotation_processors,
+        transitive_runtime_jars = transitive_runtime_jars,
+        plugins = plugins,
+        outputs = {
+            "ksp_generated_java_srcjar": ksp_generated_java_srcjar,
```

Comment:
This action currently only supports collecting generated Kotlin source files and resources. If a KSP plugin outputs Java or compiled outputs, this action will not collect them up to be used in the downstream actions.